Cpp_test/lib/file-handler.h

```diff
@@ -0,0 +1,33 @@
+#ifndef FILE_HANDLER_H_
+#define FILE_HANDLER_H_
+
+#include <iostream>
+#include <fstream>
+#include <utility>
+#include <boost/filesystem.hpp>
+
+enum class FileMode
+{
+	NONE, READ, WRITE
+};
+
+class FileHandler
+{
+	protected:
+		std::fstream fs{};
+		char *file_name;
+		FileMode mode{FileMode::NONE};
```

Comment:
As you specify a constructor, I think you should avoid initialize the variables in their declaration.